### ISSUE TITLE
Shouldn't be QQ.algebraic_field(I) == QQ.frac_field(I)?

### ISSUE THREAD

**Issue by skirpichev** [sympy/sympy#14349]
FracField doesn't assume any algebraic relations among generators, yet in similar cases it sounds odd if we will consider ``I`` just as a symbol, ignoring ``I**2 == -1``.

**Comment by jksuom** [comment]
`QQ.frac_field(I)`, as currently implemented, is clearly wrong. I think its implementation should consist of a `FracField` with no generators and `domain=QQ(I)`, where `QQ(I)` denotes the fraction field of Gaussian integers. It would be isomorphic to `QQ(I)` but more complicated and therefore not very useful.

`QQ(I)` is currently implemented as`QQ.algebraic_field(I)`, but there could probably be a more efficient implementation as a free algebra of rank two over `QQ`. That could be implemented in the same way as `Quaternion`. Then `construct_domain` could be extended to automatically [include `I` in the coefficient domain](https://github.com/sympy/sympy/blob/master/sympy/polys/constructor.py#L22) even if `algebraic=False`. If that were done, then `QQ.frac_field(I)` would have `domain=QQ(I)` (and no generators) in its `FracField`.

**Comment by skirpichev** [comment]
@jksuom, to be more precise, the whole issue is not just about ``QQ(I)``.  I think, that same is valid for ``QQ.algebraic_field(sqrt(2))`` vs ``QQ.frac_field(sqrt(2))`` and so on.

**Comment by jksuom** [comment]
I agree. `QQ.frac_field(sqrt(2)` should be implemented in the way with `domain=QQ(sqrt(2))`.
